src/main/java/com/fvv/bookstore/controller/BookControllerImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.fvv.bookstore.controller;
+
+import java.util.List;
+
+import com.fvv.bookstore.bean.Book;
+import com.fvv.bookstore.dao.BookDAO;
+import com.fvv.bookstore.dao.BookDAOImpl;
+import com.fvv.bookstore.exception.ControllerException;
+import com.fvv.bookstore.exception.DaoException;
+
+/**
+ * Controller class for a Book object.
+ * 
+ * @author Fatima Vasquez
+ * Created on 24 de mar de 2018	
+ * @version 1.0 	
+ *
+ */
+public class BookControllerImpl implements BookController {
+	
+	private BookDAO bookDao;
+	
+	/**
+	 * Class constructor instantiating a new BookDAOImpl object.
+	 */
+	public BookControllerImpl() {
+		this.bookDao = new BookDAOImpl(); 
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public boolean addBook(Book book) throws ControllerException {
+		try {
+			return this.bookDao.addBook(book);
+		} catch (DaoException e) {
+			throw new ControllerException("Error to add a book", e);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public List<Book> listBooks() throws ControllerException {
+		List<Book> books;
```

Comment:
you dont need this variable here, you can simply:

```
return this.bookDao.listBooks();
```

Reply:
OK!